src/components/ModalWithForm.js

```diff
@@ -6,26 +6,41 @@ class ModalWithForm extends Modal {
     this._modalForm = this._modalElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit.bind(this);
     this._modalInput = this._modalForm.querySelector(".modal__input");
+    this._modalButton = this._modalElement.querySelector(".modal__button");
+  }
+
+  setButtonText(loading) {
+    if (loading) {
+      console.log("saving");
+      this._modalButton.textContent = "SAVING...";
+    }
+    if (loading === false) {
+      this._modalButton.textContent = "SAVE";
+    }
   }
 
   close() {
-    this._modalForm.reset();
     super.close();
   }
 
+  _clearForm() {
+    this._modalForm.reset();
+  }
+
   _getInputValues() {
-    const inputList = [...this._modalForm.querySelectorAll('input')];
+    const inputList = [...this._modalForm.querySelectorAll("input")];
     const inputValues = {};
     for (const input of inputList) {
       inputValues[input.name] = input.value;
-    };
+    }
     return inputValues;
   }
 
   setEventListeners() {
     this._modalForm.addEventListener("submit", (e) => {
       e.preventDefault();
       this._handleFormSubmit(this._getInputValues());
+      this._clearForm();
```

Comment:
You need to **reset** the form only  in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.